test/unit/authorizer_test.rb

```diff
@@ -255,4 +255,39 @@ def setup
     assert_includes results, host
     refute results.grep(host).first.readonly?
   end
+
+  test "#find_collection(Host, :permission => :view_hosts, :joined_on: Report) for matching unlimited filter" do
```

Comment:
Perhaps put these tests on the same context with a setup block that sets `permission`